survey/dashboards/callbacks.py

```diff
@@ -0,0 +1,308 @@
+        section_averages = calculate_section_averages(filtered_responses, dash_app)
+
+        return {
+            "section_averages": section_averages,
+            "survey_responses": filtered_responses,
+        }
+
+    @dash_app.callback(
+        [Output(f'{field["id"]}-filter', "options") for field in demographic_fields],
+        [Input("stored-data", "data")],
+    )
+    def update_filter_options(data):
+        options = []
+
+        for field in demographic_fields:
+            if field["id"] == "age":
+                options.append([
+                    {"label": opt, "value": opt}
+                    for opt in ["Under 25", "25-34", "35-44", "45-54", "55-64", "65+"]
+                ])
+            else:
+                options.append([
+                    {"label": opt, "value": opt}
+                    for opt in field["options"]
+                ])
+
+        return options
+
+    # Clear filters callback
+    @dash_app.callback(
+        [Output(f'{field["id"]}-filter', "value") for field in demographic_fields],
+        [Input("clear-filters", "n_clicks")],
+        prevent_initial_call=True,
+    )
+    def clear_filters(n_clicks):
+        return [None] * len(demographic_fields)
+
+    # Metrics callbacks
+    @dash_app.callback(
+        [
+            Output("total-responses", "children"),
+            Output("completion-rate", "children"),
+            Output("average-score", "children"),
+        ],
+        [Input("stored-data", "data")],
+    )
+    def update_metrics(data):
+        if not data or "survey_responses" not in data:
+            return "0", "0%", "0.0"
+
+        try:
+            total_responses = len(data["survey_responses"])
+            if total_responses == 0:
+                return "0", "0%", "0.0"
+
+            completed = sum(
+                1
+                for response in data["survey_responses"]
+                if "answers" in response and len(response["answers"]) >= len(sections)
+            )
+
+            completion_rate = completed / total_responses * 100
+
+            all_scores = []
+
+            for section in sections:
+                section_stats = calculate_section_statistics(data, section["index"], dash_app)
+                if section_stats:
+                    mean_score = float(
+                        next(
+                            (stat["value"] for stat in section_stats if stat["statistic"] == "Mean Score"),
+                            0
+                        )
+                    )
+                    all_scores.append(mean_score)
+
+            avg_score = sum(all_scores) / len(all_scores) if all_scores else 0
+
+            return (
+                f"{total_responses:,}",
+                f"{completion_rate:.0f}%",
+                f"{avg_score:.1f}",
+            )
+
+        except Exception as e:
+            print(f"Error in update_metrics: {str(e)}")
+            return "0", "0%", "0.0"
```

Comment:
How come we're capturing all exceptions? Exceptions should be specific for situations that can be handled, otherwise let them bubble up.